subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/ArtifactTransformsReportTask.java

```diff
@@ -0,0 +1,147 @@
+import org.gradle.internal.logging.text.StyledTextOutput;
+import org.gradle.internal.logging.text.StyledTextOutputFactory;
+import org.gradle.internal.serialization.Cached;
+import org.gradle.util.internal.ClosureBackedAction;
+import org.gradle.work.DisableCachingByDefault;
+
+import javax.inject.Inject;
+
+/**
+ * A task which reports information about the Artifact Transforms (implemented by {@link TransformAction}) used by a project.
+ *
+ * This is useful for investigating ambiguous transformation scenarios.  The output can help predict which ATs will need
+ * to be modified to remove ambiguity.  This class implements {@link Reporting} to make configuring additional file output formats simple.
+ *
+ * @since 8.12
+ */
+@Incubating
+@DisableCachingByDefault(because = "Produces only non-cacheable console output")
+public abstract class ArtifactTransformsReportTask extends DefaultTask implements Reporting<ArtifactTransformReports> {
+    private final ArtifactTransformReports reports;
+    private final Cached<ArtifactTransformReportModel> reportModel = Cached.of(() -> buildReportModel(getProject()));
+
+    @Inject protected abstract ObjectFactory getObjectFactory();
+    @Inject protected abstract StyledTextOutputFactory getTextOutputFactory();
+
+    /**
+     * Creates a new instance.
+     * @since 8.12
+     */
+    public ArtifactTransformsReportTask() {
+        reports = getObjectFactory().newInstance(ArtifactTransformReportsImpl.class);
+    }
+
+    /**
+     * Limits the report to reporting on transforms using a type with this (simple) classname.
+     *
+     * @return property holding name of the type of transform to report
+     * @since 8.12
+     */
+    @Input
+    @Optional
+    @Option(option = "type", description = "The type of Transform to report")
+    public abstract Property<String> getTransformType();
+
+    /**
+     * The reports to be generated by this task.
+     */
+    @Override
+    @Nested
+    public final ArtifactTransformReports getReports() {
+        return reports;
+    }
+
+    /**
+     * Configures the reports to be generated by this task.
+     *
+     * The contained reports can be configured by task name and closures.
+     *
+     * @param closure The configuration block for the reports
+     * @return The reports container
+     */
+    @Override
+    @SuppressWarnings("rawtypes")
+    public ArtifactTransformReports reports(@DelegatesTo(value = ArtifactTransformReports.class, strategy = Closure.DELEGATE_FIRST) Closure closure) {
+        return reports(new ClosureBackedAction<>(closure));
+    }
+
+    /**
+     * Configures the reports to be generated by this task.
+     *
+     * The contained reports can be configured by task name and closures.
+     *
+     * @param configureAction The configuration block for the reports
+     * @return The reports container
+     */
+    @Override
+    public ArtifactTransformReports reports(Action<? super ArtifactTransformReports> configureAction) {
```

Comment:
I find it odd that we are returning `reports` here. 

These `Action`-accepting methods are generally intended to be used in the DSL as configuration blocks, where we don't expect users to chain calls.

Is there a reason you are returning `reports` here?

Reply:
I see now this comes from implementing `Reporting`